### ISSUE TITLE
[locale-modal]: various a11y issues

### ISSUE THREAD

**Issue by kennylam** [carbon-design-system/carbon-for-ibm-dotcom#8860]
### Description

1. Neither of the two modal windows have a label. This results in screen readers announcing them only by role (`dialog`). A label should be added to both windows announcing their main headings.

<img width="738" alt="image" src="https://user-images.githubusercontent.com/909118/168809498-b963ace8-f8c9-4172-8361-3358d245c49f.png">

<img width="737" alt="image" src="https://user-images.githubusercontent.com/909118/168809550-a8af9540-3acf-41cd-8194-35fe84b65a65.png">

2. The location/language list is enclosed in a `role=list`. This prevents the user from using up/down keys for navigation and should be removed.

<img width="737" alt="image" src="https://user-images.githubusercontent.com/909118/168809948-21efcaa4-d34f-4968-a834-57dc7eba66e6.png">

### Component(s) impacted

`locale-modal`

### Browser

Chrome

### Carbon for IBM.com version

v1.34.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/web-components/?path=/story/components-locale-modal--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
